pkg/analysis/importalias/analyzer.go

```diff
@@ -0,0 +1,179 @@
+	inspector := pass.ResultOf[inspect.Analyzer].(*inspector.Inspector)
+	nodeFilter := []ast.Node{ // filter needed nodes: visit only them
+		(*ast.ImportSpec)(nil),
+	}
+
+	inspector.Preorder(nodeFilter, func(node ast.Node) {
+		importStmt := node.(*ast.ImportSpec)
+
+		if importStmt.Name == nil {
+			return
+		}
+		alias := importStmt.Name.Name
+		if alias == "" {
+			return
+		}
+		aliasSlice := strings.Split(alias, "_")
+		path := strings.ReplaceAll(importStmt.Path.Value, "\"", "")
+		// replace all separators with `/` for normalization
+		path = strings.ReplaceAll(path, "_", "/")
+		path = strings.ReplaceAll(path, ".", "/")
+		path = strings.ReplaceAll(path, "-", "")
+		// omit the domain name in path
+		pathSlice := strings.Split(path, "/")[1:]
+		packageName := pathSlice[len(pathSlice)-1]
+
+		if !checkVersion(aliasSlice[len(aliasSlice)-1], pathSlice) {
+			applicableAlias := getAliasFix(pathSlice)
+			pass.Report(
+				analysis.Diagnostic{
+					Pos:     node.Pos(),
+					Message: fmt.Sprintf("version not specified in alias. path: %s alias: %s version %s", path, alias, packageName),
+					SuggestedFixes: []analysis.SuggestedFix{
+						{
+							Message: fmt.Sprintf("should replace %q with %q", alias, applicableAlias),
+							TextEdits: []analysis.TextEdit{
+								{
+									Pos:     importStmt.Pos(),
+									End:     importStmt.Name.End(),
+									NewText: []byte(applicableAlias),
+								},
+							},
+						},
+					},
+				},
+			)
+			return
+		}
+		if ok, lintErrMsg := checkAliasName(aliasSlice, pathSlice, pass); !ok {
+			applicableAlias := getAliasFix(pathSlice)
+			pass.Report(
+				analysis.Diagnostic{
+					Pos:     node.Pos(),
+					Message: fmt.Sprintf(lintErrMsg+" path: %s alias: %s", path, alias),
+					SuggestedFixes: []analysis.SuggestedFix{
+						{
+							Message: fmt.Sprintf("should replace %q with %q", alias, applicableAlias),
+							TextEdits: []analysis.TextEdit{
+								{
+									Pos:     importStmt.Pos(),
+									End:     importStmt.Name.End(),
+									NewText: []byte(applicableAlias),
+								},
+							},
+						},
+					},
+				},
+			)
+			return
+		}
+	})
+	return nil, nil
+}
+
+// checkVersion checks that if package name starts with `v` it's included in alias name
+func checkVersion(aliasLastWord string, pathSlice []string) bool {
+	versionExists, versionPos := packageVersion(pathSlice)
+	if !versionExists {
+		return true
+	}
+	return aliasLastWord == pathSlice[versionPos]
+
+}
+
+// checkAliasName check consistency in alias name
+func checkAliasName(aliasSlice []string, pathSlice []string, pass *analysis.Pass) (bool, string) {
+	lastUsedWordIndex := -1
+	for _, name := range aliasSlice {
+		// we don't check version rule here
+		if strings.HasPrefix(name, "v") || name == "" {
+			continue
+		}
+		usedWordIndex := searchString(pathSlice, name)
+
+		if usedWordIndex == len(pathSlice) {
+			return false, "used words in alias most be present in path"
```

Comment:
Let's format these errors carefully to make them read more clearly for the user

```suggestion
			return fmt.Errorf("alias %q uses words that are not in path %q",
			        strings.Join(aliasSlice, "_"), strings.Join(pathSlice, "/")
```

Notice that we use %q to clarify the data that the error refers to, and the message reads as a single sentence